### PR TITLE
docs: Fix enum fields rendering; fix uint/int references

### PR DESCRIPTION
**Pull request by 1st1**

### CODE REVIEW

File: docs/internals/protocol/dataformats.rst [hunk 1]
```diff
@@ -101,12 +101,23 @@ represented as:
...
 
-The :eql:type:`std::str` values are represented as a UTF-8 encoded byte string.
+The :eql:type:`std::str` values are represented as a UTF-8 encoded byte string,
+prefixed with an ``int32`` length field:
+
+.. code-block:: c
+
+    struct String {
+        // Number of bytes.
+        int32       length;
```
Comment (elprans) [inline docs/internals/protocol/dataformats.rst:111]: Why `int32` and not `uint32`?

Comment (elprans) [reply docs/internals/protocol/dataformats.rst:111]: Actually, I don't think this is correct.  Length prefix is part of the wrapping structure, not the encoded data itself.

Comment (1st1) [reply docs/internals/protocol/dataformats.rst:111]: https://github.com/edgedb/edgedb-js/blob/master/src/codecs/text.ts#L23-L31

Comment (1st1) [reply docs/internals/protocol/dataformats.rst:111]: https://github.com/MagicStack/py-pgproto/blob/master/codecs/text.pyx#L24-L48

**Pull request merged**